File: 2.recursion/fibonacci.js
function fibonacci(n) {
  let a = 0,
    b = 1,
    temp;

  if (n === 0) return 0;

  for (let i = 2; i <= n; i++) {
    temp = a + b;
    a = b;
    b = temp;
  }
  return b;
}

console.log(fibonacci(5)); // Outputs: 5

File: fibonacci/fibonacci.py
# 1. loop

# def fibonacci(n):
#     a, b = 0, 1
#     for _ in range(n):
#         a, b = b, a + b
#     return a

# print(fibonacci(5)) # Outputs: 5

# 2. recursion
def fibonacci(n):
    if n <= 1:
        return n
    else:
        return fibonacci(n-1) + fibonacci(n-2)

print(fibonacci(5)) # Outputs: 5